### PR TITLE
Enable manual aria-label handling, deprecating aria-label generation

### PR DESCRIPTION
**Pull request by CezCz**
Developer can setup aria-label or aria-labelledby to anything in an Icon.

## Related Issue

https://github.com/adobe/coral-spectrum/issues/23

## Motivation and Context

aria-label and aria-labelledby should be always handled by end user of spectrum. Spectrum, even though it would be nice, cannot guess the need for aria-label/labelledby and contextual text associated with it.

## How Has This Been Tested?

Automated tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by icaraps** [comment]
> To be honest i don't think it was never the case for "Users to be able to provide their own" given that most people ignore accessibility, so its better to have the component set labels for things they own. That being said, the same is the case with "id" where a user given value should be respected. Instead of adding a custom coral-aria-label, i think the best way is to check if aria-label does not exist before adding it, instead of adding API surface.
@joekukish

**Comment by icaraps** [comment]
IMO i think this approach is going in the right direction. As here in the PR, I would add a new `label` property/attribute to components that are setting an `aria-label` by default  and `label` would take precedence over the default behavior. 
So e.g with Icon you could write `<coral-icon label="my label"></coral-icon>` to override the default `aria-label`. 
Same goes for e.g Tab `<coral-tab label="my label">` which would set the `aria-label` to the inner icon or `<coral-checkbox label="my label"`> which would set the `aria-label` to the inner input etc.

If you don't specify `label`, it falls back to the default behavior so no breaking change to the current behavior.

**Comment by CezCz** [comment]
The fallback idea is certainly better, than aria-labels derived from the icon name. We could fallback to title or other attributes that already exist even.

However the issue I would like to additionally address is that I don't want icon to have an aria-label at all. I want to have aria-label on parent tab element and icon to have none.

The issue that is not covered by this fallback approach is when library user wants to have full control over aria-label as there might be a visible label which then should be handled by aria-labelledby.

**Comment by icaraps** [comment]
> I don't think there's the need to reinvent an attribute (coral-aria-label) when the HTML spec already has something for this. The component should just respect it, just like we should do for User Provided IDs.
@joekukish

**Comment by CezCz** [comment]
Does it mean you opt for aria-hidden :) ?

File: coral-component-icon/src/scripts/Icon.js [hunk 1]
```diff
@@ -324,7 +334,8 @@ class Icon extends BaseComponent(HTMLElement) {
    */
   _updateAltText(value) {
     const isImage = this.contains(this._elements.image);
-    
+    const shouldHandleAriaLabel = this.getAttribute(AUTO_ARIA_LABEL, 'on') !== 'off';
```
Comment (icaraps) [inline coral-component-icon/src/scripts/Icon.js]: What's the second param for `getAttribute` supposed to do ? See https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttribute

Comment (CezCz) [reply coral-component-icon/src/scripts/Icon.js]: I thought that's the default? Apparently not

**Comment by CezCz** [comment]
I thought https://github.com/adobe/coral-spectrum/pull/27/files#diff-02aaf4a26a2053b8b98797b25eb56fedR409 is checking if it's on by default. Isn't it?

**Comment by CezCz** [comment]
closing as it was integrated in commit 2d2649ce4cc90a4c4d4360c8edcefdfdae4e4ec1 in master

**Pull request closed without merging**